File: src/ru/org/linux/site/BadParameterException.java
package ru.org.linux.site;

public class BadParameterException extends ScriptErrorException
{
	public BadParameterException(String param, String info)
	{
		super("Неправильный формат параметра ``"+param+"'': "+info);
	}

	public BadParameterException(String param)
	{
		super("Неправильный формат параметра ``"+param+"''");
	}

}

File: src/main/scala/ru/org/linux/site/BadParameterException.scala
/*
 * Copyright 1998-2026 Linux.org.ru
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */
package ru.org.linux.site

class BadParameterException(param: String, info: String | Null)
    extends ScriptErrorException(
      if info != null then
        s"Неправильный формат параметра ``$param'': $info"
      else
        s"Неправильный формат параметра ``$param''"):
  def this(param: String) = this(param, null)
